lib/ConsumerImpl.cc

```diff
@@ -1186,6 +1186,7 @@ void ConsumerImpl::closeAsync(ResultCallback originalCallback) {
     if (ackGroupingTrackerPtr_) {
         ackGroupingTrackerPtr_->close();
     }
+    negativeAcksTracker_.close();
```

Comment:
The `shutdown` already closed it. We don't need to close it again here. The call of `shutdown` is at line 1165

Reply:
These two invoke scenarios are not the same:

- `consumer.close()`: This must be closed before sending the `newCloseConsumer` command to prevent the `tracker` from sending useless requests.
- `consumer.shutdown()`: In addition to the method called by consumer.close, the destructor also calls this method. When the consumer `subscriptio`n fails, the consumer object destruction is triggered, and the `tracker` needs to be closed. This one is similar to this issue: https://github.com/apache/pulsar-client-cpp/pull/185

Reply:
Make sense to me.